Bottom: c61b24147b53d5633133ac638ef68a5ff802b44d
Top:    fae0ddc79a572f1d4b19900cba92b2b54d150374
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 15:26:53 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index f7cca85..3b35891 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -803,7 +803,7 @@ update_offset_entry(struct autofs_point *ap, const char *name,
 
 	mc = source->mc;
 
-	if (!*path || !*loc) {
+	if (!*path) {
 		error(ap->logopt,
 		      MODPREFIX "syntax error in offset %s -> %s", path, loc);
 		return CHE_FAIL;
@@ -837,8 +837,10 @@ update_offset_entry(struct autofs_point *ap, const char *name,
 	if (*myoptions) {
 		strcpy(m_mapent, "-");
 		strcat(m_mapent, myoptions);
-		strcat(m_mapent, " ");
-		strcat(m_mapent, loc);
+		if (loc) {
+			strcat(m_mapent, " ");
+			strcat(m_mapent, loc);
+		}
 	} else
 		strcpy(m_mapent, loc);
